boule/ellipsoid.py

```diff
@@ -76,6 +76,17 @@ class Ellipsoid:
     long_name = attr.ib(default=None)
     reference = attr.ib(default=None)
 
+    @flattening.validator
+    def check_flattening(self, flattening, value):  # pylint: disable=no-self-use
+        """
+        Check if flattening is not equal (or almost) zero
+        """
+        if value < 1e-7:
```

Comment:
This is the sort of thing that tends to come back to bite us in the future 🙂 

Could this be a warning instead saying that some calculations could result in errors if the flattening is too close to zero (and to use Sphere instead)? Then it's up to the user to ignore it at their own peril.

Reply:
There could be a separate check to see if flattening is >= 0 though raising an error.

Reply:
Might be good to open an issue to add checks to all parameters actually.

Reply:
I was unsure about this error and with the value of the threshold... And I agree, your idea is way better. I'll open an issue for parameter checks. We also want the flattening to be less (or equal, I think...) than 1.